src/python/grpcio_tests/tests/interop/xds_interop_client.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2020 The gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import signal
+import threading
+import time
+import sys
+
+from typing import DefaultDict, List, Set
+import collections
+
+from concurrent import futures
+
+import grpc
+
+from src.proto.grpc.testing import test_pb2
+from src.proto.grpc.testing import test_pb2_grpc
+from src.proto.grpc.testing import messages_pb2
+from src.proto.grpc.testing import empty_pb2
+
+
+# TODO: Back with a LoadBalancerStatsResponse proto?
+class _StatsWatcher:
+    _start: int
+    _end: int
+    _rpcs_needed: int
+    _rpcs_by_peer: DefaultDict[str, int]
+    _no_remote_peer: int
+    _lock: threading.Lock
+    _condition: threading.Condition
```

Comment:
nit: `threading.Condition` creates a lock if got given. We could use that one `with self._condition:`.